frontend/lib/about/dev_settings_view.dart

```diff
@@ -179,6 +164,53 @@ class DevSettingsView extends StatelessWidget {
     );
   }
 
+  Future<void> _activateCard(BuildContext context, String base64qrcode) async {
+    final messengerState = ScaffoldMessenger.of(context);
+    final provider = Provider.of<UserCodeModel>(context, listen: false);
+    final client = GraphQLProvider.of(context).value;
+    final projectId = Configuration.of(context).projectId;
+    try {
+      final activationCode =
+          const ActivationCodeParser().parseQrCodeContent(const Base64Decoder().convert(base64qrcode));
+
+      final activationResult = await activateCode(
+        client: client,
+        projectId: projectId,
+        code: activationCode,
+        overwriteExisting: true,
+      );
+
+      if (activationResult.activationState == ActivationState.failed) {
+        throw QrCodeParseException("Code could not be activated from the server");
+      }
+      if (activationResult.activationState == ActivationState.didNotOverwriteExisting) {
+        throw QrCodeParseException("Did not overwrite existing QrCode.");
+      }
+      if (activationResult.totpSecret != null) {
```

Comment:
If activationResult.totpSecret is null, an error should be shown to the user.

Reply:
This is in the dev_settings_view. Are you sure we need an error here? In the actual activationWorkflow I *do* show an error.

Reply:
It would still be good to throw an error in this case. Alternatively, just remove the `if` check and let it throw automatically when you dereference `activationResult.totpSecret!`. It's better not to show a success snackbar, if there was no success :)